lib/rust/ensogl/component/slider/src/model.rs

```diff
@@ -68,16 +75,75 @@ mod track {
-            let track = Rect((&width * &slider_fraction_filled,&height));
-            let track = track.translate_x(&width * (&slider_fraction_filled - 1.0) * 0.5);
+            let track = Rect((
+                &width * &slider_fraction_horizontal,
+                &height * &slider_fraction_vertical,
+            ));
+            let track = track.translate_x(&width * (&slider_fraction_horizontal - 1.0) * 0.5);
+            let track = track.translate_y(&height * (&slider_fraction_vertical - 1.0) * 0.5);
             let track = track.intersection(background).fill(color);
             track.into()
         }
     }
 }
 
+/// Thumb shape that moves along the slider proportional to the slider value.
+mod thumb {
+    use super::*;
+
+    ensogl_core::shape! {
+        above = [background];
+        pointer_events = false;
+        (style:Style, slider_fraction:f32, thumb_width:f32, thumb_height:f32, color:Vector4) {
+            let Background{width,height,shape: background} = Background::new();
+            let thumb_width = &width * &thumb_width;
+            let thumb_height = &height * &thumb_height;
+            let thumb = Rect((&thumb_width, &thumb_height));
+            let thumb = thumb.corners_radius(&thumb_height / 2.0);
+            let range_x = &width - &thumb_width;
+            let range_y = &height - &thumb_height;
+            let thumb = thumb.translate_x(-&range_x * 0.5 + &range_x * &slider_fraction);
+            let thumb = thumb.translate_y(-&range_y * 0.5 + &range_y * &slider_fraction);
+            let thumb = thumb.intersection(background).fill(color);
+            thumb.into()
+        }
+    }
+}
+
+/// Triangle shape used as an overflow indicator on either side of the range.
+mod overflow {
+    use super::*;
+
+    ensogl_core::shape! {
+        above = [background, track, thumb];
+        pointer_events = false;
+        (style:Style, color:Vector4) {
+            let width: Var<Pixels> = "input_size.x".into();
+            let height: Var<Pixels> = "input_size.y".into();
+            let width = width - COMPONENT_MARGIN.px() * 2.0;
+            let height = height - COMPONENT_MARGIN.px() * 2.0;
+
+            let color = style.get_color(theme::overflow::color);
+            let triangle = Triangle(width.clone(), height.clone());
+            let triangle = triangle.fill(color);
+
+            let point_a = (-width.clone() / 2.0, -height.clone() / 2.0);
+            let point_b = (width / 2.0, -height.clone() / 2.0);
+            let point_c = (0.0.px(), height / 2.0);
+            let segment_ab = Segment(point_a.clone(), point_b.clone(), 3.0.px());
+            let segment_bc = Segment(point_b, point_c.clone(), 3.0.px());
+            let segment_cd = Segment(point_c, point_a, 3.0.px());
+            let triangle = triangle.union(segment_ab);
+            let triangle = triangle.union(segment_bc);
+            let triangle = triangle.union(segment_cd);
```

Comment:
what are these segments for? Is triangle shape not visible, or something else happened?

Reply:
That was my attempt at a triangle shape with rounded corners. If I've missed a more convenient way to do that I'd be happy to change this.